src/modules/views/Folder/hooks/useExtraDrive.ts

```diff
@@ -1,41 +1,30 @@
-import { useMemo } from 'react'
-
 import { useQuery } from 'cozy-client'
 
-import { useFolderSort } from 'modules/navigation/duck'
-import { buildDriveQuery } from 'modules/queries'
-
-interface File {
-  _id: string
-  path: string
-  attributes: {
-    name: string
-  }
-}
-
-export const useExtraDrive = (): File[] | undefined => {
-  const folderId = '3130bb7c264db92cf1955f62b600097a'
-  const [sortOrder] = useFolderSort(folderId) as [
-    { attribute: string; order: string }
-  ]
-
-  const buildedFilesQuery = buildDriveQuery({
-    currentFolderId: folderId,
-    type: 'file',
-    sortAttribute: sortOrder.attribute,
-    sortOrder: sortOrder.order
-  })
-
-  const filesQuery = useQuery(
-    buildedFilesQuery.definition(),
-    buildedFilesQuery.options
-  ) as { data: File[] | undefined }
-
-  const files = useMemo(() => {
-    if (filesQuery.data?.length === 0) return undefined
-
-    return filesQuery.data
-  }, [filesQuery.data])
-
-  return files
+import { buildExternalDriveQuery } from 'modules/views/Folder/queries/fetchExtraDrive'
+import { ExtraDriveFile } from 'modules/views/Folder/types'
+
+/**
+ * Custom React hook to fetch data from an external drive folder.
+ *
+ * This hook encapsulates the logic for building the query parameters and
+ * executing the query to fetch files from an external drive. It leverages
+ * the 'cozy-client' library's useQuery hook for data fetching.
+ *
+ * @returns An array of ExtraDriveFile objects representing the files fetched
+ * from the external drive. If no data is fetched, an empty array is returned (be careful to handle this case).
+ */
+export const useExtraDrive = (): ExtraDriveFile[] => {
+  // Build the query parameters for fetching external drive data
+  const query = buildExternalDriveQuery()
+
+  // Execute the query using cozy-client's useQuery hook
+  // The result of the query is cast to ensure it's treated as an array of ExtraDriveFile objects
+  const { data } = useQuery(query.definition, query.options) as
+    | {
+        data: ExtraDriveFile[]
+      }
+    | { data: null }
+
+  // Return the fetched data
+  return data ?? []
```

Comment:
nit: Maybe return the data into an object to be able in future to add a loading state, for example, but this may be premature optimisation

Reply:
What is the point to have this custom hook instead of just using the build method and the query directly within the component? You are creating a new API, in fact a subset API, that will bring more complexity over the time than just using the hook from cozy-client directly.

Reply:
@Crash-- it's because I used it at two different places and wanted to centralise the query at one place only. Indeed it has the side effect of hiding cozy-client here, so you would useQuery with the same definition/options at the two different components then?

Reply:
This is what we do yes. We write the buildQueryXXX, and then we call the useQuery with this query in several place.

Reply:
removed the file and did that instead 👍